src/libsync/clientsideencryption.cpp

```diff
@@ -1754,6 +1754,47 @@ QVector<EncryptedFile> FolderMetadata::files() const {
     return _files;
 }
 
+bool FolderMetadata::isFileDropPresent() const
+{
+    return _fileDrop.size() > 0;
+}
+
+bool FolderMetadata::moveFileDropToFiles()
+{
+    if (_fileDrop.isEmpty()) {
+        return false;
+    }
+
+    for (auto it = _fileDrop.constBegin(); it != _fileDrop.constEnd(); ++it) {
+        const auto fileObject = it.value().toObject();
+
+        const auto encryptedFile = fileObject["encrypted"].toString().toLocal8Bit();
+        const auto decryptedFile = decryptData(encryptedFile);
+        const auto decryptedFileDocument = QJsonDocument::fromJson(decryptedFile);
+        const auto decryptedFileObject = decryptedFileDocument.object();
+
+        EncryptedFile file;
+        file.encryptedFilename = it.key();
+        file.metadataKey = fileObject["metadataKey"].toInt();
+        file.authenticationTag = QByteArray::fromBase64(fileObject["authenticationTag"].toString().toLocal8Bit());
+        file.initializationVector = QByteArray::fromBase64(fileObject["initializationVector"].toString().toLocal8Bit());
+
+        file.originalFilename = decryptedFileObject["filename"].toString();
+        file.encryptionKey = QByteArray::fromBase64(decryptedFileObject["key"].toString().toLocal8Bit());
+        file.mimetype = decryptedFileObject["mimetype"].toString().toLocal8Bit();
+        file.fileVersion = decryptedFileObject["version"].toInt();
+
+        // In case we wrongly stored "inode/directory" we try to recover from it
+        if (file.mimetype == QByteArrayLiteral("inode/directory")) {
+            file.mimetype = QByteArrayLiteral("httpd/unix-directory");
+        }
```

Comment:
what is the story behind this ?
is this somehow documented somewhere in the E2EE API ?

Reply:
@mgallien got absolutely no idea what this means, I copy-pasted it from `void FolderMetadata::setupExistingMetadata(const QByteArray& metadata)`